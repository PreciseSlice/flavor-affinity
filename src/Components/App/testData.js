/* eslint-disable */
export const allIngredients = [
  {
    id: 5902,
    name: '1724 Tonic',
    product: '1724 Tonic',
    preparation: '',
    _links: {
      preparation: null,
      origin: null,
      image: {
        id: 4578,
        src: 'https://pantry.foodpairing.com/apps/4578',
        size_240: 'https://pantry.foodpairing.com/apps/4578_240_0',
        size_480: 'https://pantry.foodpairing.com/apps/4578_480_0',
        size_720: 'https://pantry.foodpairing.com/apps/4578_720_0',
        size_960: 'https://pantry.foodpairing.com/apps/4578_960_0',
        size_1200: 'https://pantry.foodpairing.com/apps/4578_1200_0',
        size_1440: 'https://pantry.foodpairing.com/apps/4578_1440_0'
      },
      brand: null,
      categories: [
        {
          id: 38,
          href:
            'https://api.foodpairing.com/taxonomies/ingredients/categories/38'
        },
        {
          id: 139,
          href:
            'https://api.foodpairing.com/taxonomies/ingredients/categories/139'
        }
      ]
    },
    description:
      'The essence of 1724 Tonic Water has its origins in the Andes, on the mythical Inca Trail. This is not by chance. This is where quinine was discovered. There, this ingredient gains the authenticity of a product with roots in the region that go back hundreds of years. At 1724 metres above sea level. Not a metre above or below.',
    _meta: {
      created_at: '',
      modified_at: '',
      index: 0
    }
  },
  {
    id: 5440,
    name: 'A la Chimay Bleue',
    product: 'A la Chimay Bleue',
    preparation: '',
    _links: {
      preparation: null,
      origin: null,
      image: {
        id: 1772,
        src: 'https://pantry.foodpairing.com/apps/1772',
        size_240: 'https://pantry.foodpairing.com/apps/1772_240_0',
        size_480: 'https://pantry.foodpairing.com/apps/1772_480_0',
        size_720: 'https://pantry.foodpairing.com/apps/1772_720_0',
        size_960: 'https://pantry.foodpairing.com/apps/1772_960_0',
        size_1200: 'https://pantry.foodpairing.com/apps/1772_1200_0',
        size_1440: 'https://pantry.foodpairing.com/apps/1772_1440_0'
      },
      brand: {
        id: 80,
        name: 'Chimay',
        website: '',
        description: '',
        _links: {
          images: [
            {
              id: 4420,
              src: 'https://pantry.foodpairing.com/apps/4420'
            }
          ]
        }
      },
      categories: [
        {
          id: 265,
          href:
            'https://api.foodpairing.com/taxonomies/ingredients/categories/265'
        },
        {
          id: 266,
          href:
            'https://api.foodpairing.com/taxonomies/ingredients/categories/266'
        }
      ]
    },
    description:
      'The rind of the à la Chimay Bleue is regularly washed with the Chimay Blue Cap during his stay in the aging cellars of the monestarey. Pronounced flavor with a slight bitterness of the hops.',
    _meta: {
      created_at: '',
      modified_at: '',
      index: 1
    }
  },
  {
    id: 5443,
    name: 'A la Chimay Rouge',
    product: 'A la Chimay Rouge',
    preparation: '',
    _links: {
      preparation: null,
      origin: null,
      image: {
        id: 1773,
        src: 'https://pantry.foodpairing.com/apps/1773',
        size_240: 'https://pantry.foodpairing.com/apps/1773_240_0',
        size_480: 'https://pantry.foodpairing.com/apps/1773_480_0',
        size_720: 'https://pantry.foodpairing.com/apps/1773_720_0',
        size_960: 'https://pantry.foodpairing.com/apps/1773_960_0',
        size_1200: 'https://pantry.foodpairing.com/apps/1773_1200_0',
        size_1440: 'https://pantry.foodpairing.com/apps/1773_1440_0'
      },
      brand: {
        id: 80,
        name: 'Chimay',
        website: '',
        description: '',
        _links: {
          images: [
            {
              id: 4420,
              src: 'https://pantry.foodpairing.com/apps/4420'
            }
          ]
        }
      },
      categories: [
        {
          id: 265,
          href:
            'https://api.foodpairing.com/taxonomies/ingredients/categories/265'
        },
        {
          id: 266,
          href:
            'https://api.foodpairing.com/taxonomies/ingredients/categories/266'
        }
      ]
    },
    description:
      'The à la Chimay Rouge is a round, smooth cheese with creamy texture. Its natural rind is washed with Chimay Red Cap. Fruity flavour with slight bitterness.',
    _meta: {
      created_at: '',
      modified_at: '',
      index: 2
    }
  }
];

export const cleanData = [
    {
      id: 5902,
      name: '1724 tonic',
      image: 'https://pantry.foodpairing.com/apps/4578_240_0',
      description:
        'The essence of 1724 Tonic Water has its origins in the Andes, on the mythical Inca Trail. This is not by chance. This is where quinine was discovered. There, this ingredient gains the authenticity of a product with roots in the region that go back hundreds of years. At 1724 metres above sea level. Not a metre above or below.'
    },
    {
      id: 5440,
      name: 'a la chimay bleue',
      image: 'https://pantry.foodpairing.com/apps/1772_240_0',
      description:
        'The rind of the à la Chimay Bleue is regularly washed with the Chimay Blue Cap during his stay in the aging cellars of the monestarey. Pronounced flavor with a slight bitterness of the hops.'
    },
    {
      id: 5443,
      name: 'a la chimay rouge',
      image: 'https://pantry.foodpairing.com/apps/1773_240_0',
      description:
        'The à la Chimay Rouge is a round, smooth cheese with creamy texture. Its natural rind is washed with Chimay Red Cap. Fruity flavour with slight bitterness.'
    }
];
