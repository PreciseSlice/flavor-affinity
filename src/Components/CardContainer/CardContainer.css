.card-container {
  background: rgba(87, 87, 87, 0.6);
  height: 70%;
  margin-top: 3%;
  width: 90%;
  display: flex;
  flex-direction: row;
  overflow: auto;
  flex-wrap: wrap;
  justify-content: space-evenly; }

.card-container::-webkit-scrollbar {
  width: 1em; }

.card-container::-webkit-scrollbar-track {
  -webkit-box-shadow: inset 0 0 6px rgba(0, 0, 0, 0.3); }

.card-container::-webkit-scrollbar-thumb {
  background-color: darkgrey;
  outline: 1px solid slategrey; }
