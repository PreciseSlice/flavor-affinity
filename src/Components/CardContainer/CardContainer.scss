.card-container {
  background: rgba(87, 87, 87, 0.6);
  box-shadow: 0 0 5px 5px  rgba(87, 87, 87, 0.6);
  color: white;
  height: 55%;
  margin-top: 30px; 
  width: 90%;
  display: flex;
  flex-direction: row;
  overflow: auto;
  flex-wrap: wrap;
  justify-content: space-evenly;

  h1 {
    font-size: 4em;
  }
}

.card-container::-webkit-scrollbar {
  width: 1em;
}

.card-container::-webkit-scrollbar-track {
  -webkit-box-shadow: inset 0 0 6px rgba(0,0,0,0.3);
}

.card-container::-webkit-scrollbar-thumb {
background-color: darkgrey;
outline: 1px solid slategrey;
}

.pairing-container {
  width: 100%;
  display: flex;
  margin-top: 30px;
  justify-content: center;
  align-items: center;
}

.five-container {
  border: 5px solid darkgray;
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: center;
  overflow: auto;
  width: 30%;
  margin-left: 30px;

  .card {
    height: 18%;
    width: 75%;
    margin-top: 30px;

    h1 {
      font-size: 2em;
    }

    button {
      margin-left: none; 
    }
  }
}

.star-container {
  width: 100%;
  margin-top: 30px;
  display: flex;
  align-items: center;
  justify-content: center;

  img {
    height: 60px;
    margin-left: 15px;
  }
}

.all-pairings-btn-container {
  align-items: center;
  display: flex;
  justify-content: center;
  width: 100%;
}

.all-pairings-btn{
  width: 15%;
}

.all-pairings-container {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  width: 100%;
}

.loading-container {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;

  h1 {
    margin-bottom: 30px;
  }

  img {
    box-shadow: 0 0 5px 5px  rgba(87, 87, 87, 0.6);
    height: 400px;
  }
}
