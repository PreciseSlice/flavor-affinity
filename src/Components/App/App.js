import React, { Component } from 'react';
import { connect } from 'react-redux';
import './App.css';
import MainForm from '../MainForm/MainForm';
import CardContainer from '../CardContainer/CardContainer';
//import { getAllIngredients } from '../../Helpers/apiCalls';
import { setAllIngredients } from '../../Actions';
//import * as actions from '../../Actions';
import { allDataClean } from '../../Helpers/.mockApi.js';
//import { cleanData } from '../App/testData';

export class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      error: null,
      errorInfo: null
    };
  }

  async componentDidMount() {
    const { setAllIngredients } = this.props;
    //const allIngredients = await getAllIngredients();
    //setAllIngredients(allIngredients);
    setAllIngredients(allDataClean);
  }

  componentDidCatch(error, errorInfo) {
    this.setState({
      error: error,
      errorInfo: errorInfo
    });
  }

  clearError() {
    this.setState({
      error: null,
      errorInfo: null
    });
  }

  render() {
    if (this.state.error) {
      return (
        <div className="eror">
          <h2>{'Oh-no! Something went wrong'}</h2>
          <p className="">{this.state.error && this.state.error.toString()}</p>
          <div>{'Component Stack Error Details: '}</div>
          <p className="">{this.state.errorInfo.componentStack}</p>
          <button onClick={() => this.clearError()} />
        </div>
      );
    }
    return (
      <div className="app">
        <MainForm />
        <CardContainer />
      </div>
    );
  }
}

// export const mapStateToProps = state => ({

// });

export const mapDispatchToProps = dispatch => ({
  setAllIngredients: ingredient => dispatch(setAllIngredients(ingredient))
});

export default connect(null, mapDispatchToProps)(App);
