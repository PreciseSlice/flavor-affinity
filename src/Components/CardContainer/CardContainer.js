import React, { Component } from 'react';
import { connect } from 'react-redux';
import './CardContainer.css';
import Card from '../Card/Card';
import PropTypes from 'prop-types';

class CardContainer extends Component {
  render() {
    const { allIngredients } = this.props;
    
    if (allIngredients) {
      const renderCards = allIngredients.map(ingredient => {
        return <Card data={ingredient} key={ingredient.id} />;
      });

      return <div className="card-container">{renderCards}</div>;
    } else {
      return null;
    }
  }
}

export const mapStateToProps = state => ({
  allIngredients: state.ingredients
});

export default connect(mapStateToProps, null)(CardContainer);

CardContainer.propTypes = {
  allIngredients: PropTypes.oneOfType([PropTypes.array, PropTypes.arrayOf(
    PropTypes.shape({
      description: PropTypes.string,
      id: PropTypes.number.isRequired,
      image: PropTypes.string.isRequired,
      name: PropTypes.string.isRequired
    })
  )]),
};
