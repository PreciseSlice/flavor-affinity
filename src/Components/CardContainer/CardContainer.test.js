/* eslint-disable */
import React from 'react';
import { shallow } from 'enzyme';
import { CardContainer, mapStateToProps } from './CardContainer';
import { cleanData } from '../App/testData';

describe('CardContainer', () => {

  const pairingsObject = {
    topFive: []
  }

  it('matches snapshot', () => {
    const wrapper = shallow(
      <CardContainer
        allIngredients={cleanData}
        suggestedIngredients={cleanData}
        pairingsObject={pairingsObject}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should map items in the store to props', () => {
    const mockStore = {
      state: {
        ingredients: cleanData,
        suggestedIngredients: cleanData,
        pairingsObject: {}
      }
    };
    const mapped = mapStateToProps(mockStore);

    expect(mapped.allIngredients).toEqual(mockStore.ingredients);
    expect(mapped.suggestedIngredients).toEqual(mockStore.suggestedIngredients);
    expect(mapped.pairingsObject).toEqual(mockStore.pairingsObject);
  });
});
