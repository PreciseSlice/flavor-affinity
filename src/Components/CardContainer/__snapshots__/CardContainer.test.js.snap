// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContainer exist and matches snapshot 1`] = `<Connect(Component) />`;
